src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -166,15 +166,8 @@ function _M:getTokenFromCache(cacheLookupKey)
     return nil;
 end
 
--- imsAuth will validate the service token passed in "Authorization" header --
-function _M:validate_ims_token()
+function _M:validateOAuthToken(oauth_token, validation_config)
```

Comment:
if `validation_config` is `nil` or `validation_config.RESPONSES` is `nil` this method will fail with NPE.
We could either create a getter for `validation_config` that can be used in other methods too, or make sure that it's not breaking this method.
The getter could be:
```
local function get_validation_config( validation_config )
    local cfg = validation_config or {}
    cfg.RESPONSES = validation_config.RESPONSES or RESPONSES
    return cfg
end
```

WDYT ?